libsolidity/formal/SMTChecker.cpp

```diff
@@ -752,6 +759,7 @@ void SMTChecker::mergeVariables(vector<VariableDeclaration const*> const& _varia
 	set<VariableDeclaration const*> uniqueVars(_variables.begin(), _variables.end());
 	for (auto const* decl: uniqueVars)
 	{
+		solAssert(_countersEndTrue.count(decl) && _countersEndFalse.count(decl), "");
```

Comment:
Do we want to be explicit about the check? The `at`s below already throw an exception if `decl` is not there.

Reply:
The `.at` will throw a `std::map` exception which is not understood by our exception guards and it just ends up being the unexpressive "Unknown exception during compilation" text.